spec/ad.ts

```diff
@@ -0,0 +1,430 @@
+            .set('content-type', 'application/json')
+            .send({
+              filter: {
+                kmsFrom: 1000,
+                markId
+              }
+            });
+          chai.assert.fail('Test failed');
+        } catch (err) {
+          err.response.should.have.status(HttpStatus.INTERNAL_SERVER_ERROR);
+          chai.assert.equal(codeErrors.INTERNAL_DB_ERROR, err.response.body.error.code);
+        }
+        mongoStub.restore();
+      });
+      describe('Should be return failed status', () => {
+        it('if do not send required field', async () => {
+          try {
+            await chai
+              .request(app)
+              .post('/api/ad')
+              .set('content-type', 'application/json')
+              .send({});
+            chai.assert.fail('Test failed');
+          } catch (err) {
+            err.response.should.have.status(HttpStatus.UNPROCESSABLE_ENTITY);
+            chai.assert.equal(codeErrors.REQUIRED_FIELD, err.response.body.error.code);
+          }
+        });
+        it('if limit field is invalid', async () => {
+          try {
+            await chai
+              .request(app)
+              .post('/api/ad')
+              .set('content-type', 'application/json')
+              .send({
+                filter: {
+                  markId
+                },
+                limit: -1
+              });
+            chai.assert.fail('Test failed');
+          } catch (err) {
+            err.response.should.have.status(HttpStatus.UNPROCESSABLE_ENTITY);
+            chai.assert.equal(codeErrors.VALIDATION_ERROR, err.response.body.error.code);
+          }
+        });
+        it('if skip field is invalid', async () => {
+          try {
+            await chai
+              .request(app)
+              .post('/api/ad')
+              .set('content-type', 'application/json')
+              .send({
+                filter: {
+                  markId
+                },
+                skip: -1
+              });
+            chai.assert.fail('Test failed');
+          } catch (err) {
+            err.response.should.have.status(HttpStatus.UNPROCESSABLE_ENTITY);
+            chai.assert.equal(codeErrors.VALIDATION_ERROR, err.response.body.error.code);
+          }
+        });
+        it('if yearFrom field is invalid', async () => {
+          try {
+            await chai
+              .request(app)
+              .post('/api/ad')
+              .set('content-type', 'application/json')
+              .send({
+                filter: {
+                  markId,
+                  yearFrom: 100
+                }
+              });
+            chai.assert.fail('Test failed');
+          } catch (err) {
+            err.response.should.have.status(HttpStatus.UNPROCESSABLE_ENTITY);
+            chai.assert.equal(codeErrors.VALIDATION_ERROR, err.response.body.error.code);
+          }
+        });
+        it('if yearFrom field is invalid', async () => {
```

Comment:
It should be easy to read. It should say something like: It should faile if ... etc etc